src/controllers/controllerAuth.class.php

```diff
@@ -112,4 +126,120 @@ public function logOut() : void {
         session_destroy();
         header('Location: /login');
     }
+
+    
+
+    /**
+     * @brief Enregistre un utilisateur
+     * 
+     * @details La méthode register() est appelée par la route /register.
+     * Elle lit les données du formulaire envoyées en JSON et essaie de créer un utilisateur et un joueur.
+     * Si l'utilisateur et le joueur n'existent pas, la méthode créer l'utilisateur et le joueur.
+     * Si l'utilisateur existe déjà, la méthode renvoie un message d'erreur.
+     * Si le joueur existe déjà, la méthode renvoie un message d'erreur.
+     * Si l'utilisateur et le joueur sont créés avec succès, la méthode renvoie un message de confirmation.
+     * La méthode renvoie un JSON contenant un champ "success" et un champ "message".
+     * @return void
+     */
+    public function register(?string $username, ?string $email, ?string $password): void {
+        if(session_status() === PHP_SESSION_NONE) { session_start(); }
+        // Vérifier si l'email, le nom d'utilisateur et le mot de passe sont valides
+        if ($this->validateUsername($username) &&
+            $this->validateEmail($email) &&
+            $this->validatePassword($password)) {
+
+            // Initialiser la variable de message de résultat
+            $resultMessage = null;
+
+            // Hash le mot de passe
+            $hashedPassword = password_hash($password, PASSWORD_DEFAULT);
+
+
+            $userDAO = new UserDAO($this->getPdo());
+            $playerDAO = new PlayerDAO($this->getPdo());
+            
+            // Vérifier si l'utilisateur et le joueur existent
+            $existingUser = $userDAO->findByEmail($email) !== null;
+            $existingPlayer = $playerDAO->findByUsername($username) !== null;
+            
+            $resultUser = false;
+            
+            // Si l'utilisateur et le joueur n'existent pas, créer l'utilisateur
+            if (!$existingUser && !$existingPlayer)
+            {
+                $emailVerifToken = bin2hex(random_bytes(16)); // Générer un token de vérification de l'email
+                $resultUser = $userDAO->createUser($email, $hashedPassword, $emailVerifToken);
+
+                // Lien de vérification de l'email
+                $verificationLink = "/verifyEmail?token=$emailVerifToken";
```

Comment:
Utiliser une réelle route du contrôleur pour ça
```suggestion
                $verificationLink = "/verify-email/$emailVerifToken";
```